radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java

```diff
@@ -59,12 +59,17 @@ public boolean test(MessageInTransit msg) {
 			final View view = proposal.getVertex().getView();
 			final Set<BFTNode> nodesToDrop = proposalToDrop.computeIfAbsent(view, v -> {
 				List<BFTNode> nodes = new LinkedList<>(validatorSet);
-				Set<BFTNode> nextFaultySet = new HashSet<>();
-				for (int i = 0; i < faultySize; i++) {
-					BFTNode nextFaultyNode = nodes.remove(random == null ? i : random.nextInt(nodes.size()));
-					nextFaultySet.add(nextFaultyNode);
+				ImmutableSet.Builder<BFTNode> nextFaultySet = ImmutableSet.builder();
+				if (random != null) {
+					for (int i = 0; i < faultySize; i++) {
+						BFTNode nextFaultyNode = nodes.remove(random.nextInt(nodes.size()));
+						nextFaultySet.add(nextFaultyNode);
+					}
+				} else {
+					nodes.stream().limit(faultySize).forEach(nextFaultySet::add);
```

Comment:
Would be better as `nextFaultySet.addAll(nodes.subList(0, faultySize))`.  Could be shared with non-null case too, like this:
```
    final List<BFTNode> nodes = Lists.newArrayList(validatorSet);
    if (random != null) {
        Collections.shuffle(nodes, random);
    }
    final ImmutableSet<BFTNode> nextFaultySet = ImmutableSet.copyOf(nodes.subList(0, faultySize));
```